shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-common/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/classbased/ClassBasedShardingAlgorithm.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.sharding.algorithm.sharding.classbased;
+
+import com.google.common.base.Preconditions;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.shardingsphere.sharding.api.sharding.complex.ComplexKeysShardingAlgorithm;
+import org.apache.shardingsphere.sharding.api.sharding.complex.ComplexKeysShardingValue;
+import org.apache.shardingsphere.sharding.api.sharding.hint.HintShardingAlgorithm;
+import org.apache.shardingsphere.sharding.api.sharding.hint.HintShardingValue;
+import org.apache.shardingsphere.sharding.api.sharding.standard.PreciseShardingValue;
+import org.apache.shardingsphere.sharding.api.sharding.standard.RangeShardingValue;
+import org.apache.shardingsphere.sharding.api.sharding.standard.StandardShardingAlgorithm;
+
+import java.util.Collection;
+import java.util.Objects;
+import java.util.Properties;
+
+/**
+ * Class based sharding algorithm.
+ */
+public class ClassBasedShardingAlgorithm implements StandardShardingAlgorithm<Comparable<?>>, ComplexKeysShardingAlgorithm<Comparable<?>>, HintShardingAlgorithm<Comparable<?>> {
+
+    private static final String STRATEGY_KEY = "strategy";
+
+    private static final String ALGORITHM_CLASS_NAME_KEY = "algorithmClassName";
+
+    private StandardShardingAlgorithm standardShardingAlgorithm;
+
+    private ComplexKeysShardingAlgorithm complexKeysShardingAlgorithm;
+
+    private HintShardingAlgorithm hintShardingAlgorithm;
+
+    @Getter
+    private ClassBasedShardingAlgorithmStrategy strategy;
+
+    @Getter
+    private String algorithmClassName;
+
+    @Getter
+    @Setter
+    private Properties props = new Properties();
+
+    @Override
+    public void init() {
+        String strategyStr = props.getProperty(STRATEGY_KEY);
```

Comment:
strategyKey may better than strategyStr for variable name

Reply:
Yes it is, thanks